docs/README-linux.md

```diff
@@ -0,0 +1,18 @@
+# Web3.py on Linux
+
+## Developer Setup
+
+1. Install all of the package dependencies
+
+```sh
+sudo apt-get install libssl-dev libffi-dev autoconf automake libtool
+# ^ This is for Debian-like systems. TODO: Add more platforms
+
+sudo pacman -Sy libsecp256k1
+# ^ This is for ArchLinux system
+
+sudo dnf install openssl-devel libffi-devel autoconf automake libtool
+# ^ This is for Fedora.
+```
+
+2. Install `leveldb` (TODO)
```

Comment:
Have a look at this [issue](https://github.com/google/leveldb/issues/433) if it is of any help to you.
There doesn't seem to be an official guide for installing `leveldb`. In fact their [github page seems to recommend building from source](https://github.com/google/leveldb#building). 

Instead of providing OS specific installation notes, I would just provide a link to their github page. Although providing specific commands is great for a new contributor, we can't do much good when the library doesn't have an official installation guide.

However, before making any change, listen to @carver's and @pipermerriam's thoughts on this.

Reply:
Anyway, I think it's fine to just merge as is and continue iterating on the docs in new PRs. :+1: